commit 56050acb96abc7ac2079d83ffdd232056732bb47
Author: Jacob Walls <jacobtylerwalls@gmail.com>
Date:   Wed Jun 17 15:24:57 2026 -0400

    Fixed #37170 -- Applied the no-argument form of sensitive_post_parameters to MultiValueDict.

diff --git a/tests/view_tests/tests/test_debug.py b/tests/view_tests/tests/test_debug.py
index 229853b690..c31d69619f 100644
--- a/tests/view_tests/tests/test_debug.py
+++ b/tests/view_tests/tests/test_debug.py
@@ -50,6 +50,7 @@ from ..views import (
     multivalue_dict_key_error,
     non_sensitive_view,
     paranoid_view,
+    partially_sensitive_view,
     sensitive_args_function_caller,
     sensitive_kwargs_function_caller,
     sensitive_method_view,
@@ -1673,6 +1674,20 @@ class ExceptionReporterFilterTests(
             self.verify_paranoid_response(paranoid_view)
             self.verify_paranoid_email(paranoid_view)
 
+    def test_partially_sensitive_request(self):
+        """
+        No POST parameters can be seen in the default error reports for views
+        decorated with the no-argument form of sensitive_post_parameters()
+        alongside a with-arguments form of sensitive_variables().
+        """
+        with self.settings(DEBUG=True):
+            self.verify_unsafe_response(partially_sensitive_view)
+            self.verify_unsafe_email(partially_sensitive_view)
+
+        with self.settings(DEBUG=False):
+            self.verify_paranoid_response(partially_sensitive_view)
+            self.verify_paranoid_email(partially_sensitive_view)
+
     def test_multivalue_dict_key_error(self):
         """
         #21098 -- Sensitive POST parameters cannot be seen in the
diff --git a/tests/view_tests/views.py b/tests/view_tests/views.py
index 1986341177..0280e535ff 100644
--- a/tests/view_tests/views.py
+++ b/tests/view_tests/views.py
@@ -236,6 +236,24 @@ def paranoid_view(request):
         return technical_500_response(request, *exc_info)
 
 
+@sensitive_variables("cooked_eggs", "sauce")
+@sensitive_post_parameters()
+def partially_sensitive_view(request):
+    # Do not just use plain strings for the variables' values in the code
+    # so that the tests don't return false positives when the function's source
+    # is displayed in the exception report.
+    cooked_eggs = "".join(["s", "c", "r", "a", "m", "b", "l", "e", "d"])  # NOQA
+    sauce = "".join(  # NOQA
+        ["w", "o", "r", "c", "e", "s", "t", "e", "r", "s", "h", "i", "r", "e"]
+    )
+    try:
+        raise request.POST["bar"]
+    except Exception:
+        exc_info = sys.exc_info()
+        send_log(request, exc_info)
+        return technical_500_response(request, *exc_info)
+
+
 def sensitive_args_function_caller(request):
     try:
         sensitive_args_function(
